common/service-support/src/main/java/io/helidon/common/servicesupport/ServiceSupportBase.java

```diff
@@ -0,0 +1,179 @@
+     * </p>
+     *
+     * @param <T> type of the concrete service
+     * @param <B> type of the concrete builder for the service
+     */
+    public abstract static class Builder<T extends ServiceSupportBase<T, B>, B extends Builder<T, B>>
+            implements io.helidon.common.Builder<T> {
+
+        private final Class<B> builderClass;
+        private String context;
+        private Config config = Config.empty();
+        private CrossOriginConfig crossOriginConfig = null;
+
+        protected Builder(Class<B> builderClass, String defaultContext) {
+            this.builderClass = builderClass;
+            this.context = defaultContext;
+        }
+
+        /**
+         * Sets the configuration to be used by this builder.
+         * <p>
+         *     Concrete builder implementations may override this method but should invoke {@code super.config(config)} to
+         *     benefit from the common routing set-up.
+         * </p>
+         *
+         * @param config the Helidon config instance
+         * @return updated builder instance
+         */
+        public B config(Config config) {
+            this.config = config;
+
+            getWebContextConfig(config)
+                    .asString()
+                    .ifPresent(this::webContext);
+
+            config.get(CorsEnabledServiceHelper.CORS_CONFIG_KEY)
+                    .as(CrossOriginConfig::create)
+                    .ifPresent(this::crossOriginConfig);
+
+            return me();
+        }
+
+        /**
+         * Returns the config (if any) assigned for this builder.
+         *
+         * @return the Config
+         */
+        public Config config() {
+            return config;
+        }
+
+        /**
+         * Set the root context for the REST API of the service.
+         *
+         * @param path context to use
+         * @return updated builder instance
+         */
+        public B webContext(String path) {
+            if (path.startsWith("/")) {
+                this.context = path;
+            } else {
+                this.context = "/" + path;
+            }
+            return me();
+        }
+
+        /**
+         * Set the CORS config from the specified {@code CrossOriginConfig} object.
+         *
+         * @param crossOriginConfig {@code CrossOriginConfig} containing CORS set-up
+         * @return updated builder instance
+         */
+        public B crossOriginConfig(CrossOriginConfig crossOriginConfig) {
+            Objects.requireNonNull(crossOriginConfig, "CrossOriginConfig must be non-null");
+            this.crossOriginConfig = crossOriginConfig;
+            return me();
+        }
+
+        /**
+         * Returns correctly-typed {@code this}.
+         *
+         * @return typed "this"
+         */
+        private B me() {
+            return builderClass.cast(this);
+        }
+
+        protected Config getWebContextConfig(Config config) {
```

Comment:
Please do not use `get` prefix for getters.

Reply:
Fixed